File: medium/bubble-sort/solution-1.js
function bubbleSort(array) {
    let swaps = 0;
  
    for (let i = 0; i < array.length - 1; i++) {
      if (array[i] > array[i + 1]) {
        [array[i], array[i + 1]] = [array[i + 1], array[i]];
        swaps++;
      }
    }
    return swaps === 0 ? array : bubbleSort(array);
  }

File: medium/bubble-sort/solution-1.py
def bubble_sort(arr):
    # Loop through the array, starting from the beginning and going to the second-to-last element
    for i in range(len(arr) - 1):
        # Loop through the array again, starting from the end and going to the element at index `i` + 1
        for j in range(len(arr) - 1, i, -1):
            # If the element at index `j` is less than the element at index `j` - 1, swap them
            if arr[j] < arr[j - 1]:
                arr[j], arr[j - 1] = arr[j - 1], arr[j]
    # Return the sorted array
    return arr
